### DISCUSSION TITLE
Possibility of disabling events triggers with API / SDK

### DISCUSSION THREAD

**Discussion by Rafache** [directus/directus#20522] [Feature Requests]
### Summary

On the flows, we have the possibility to deactivate the triggering of events. It will be good to have the same possibility with the APIs (rest / graphQL) as well as with the SDK.

### Basic Example

I can have a flow that triggers actions on the creation of a collection item that I don't want to trigger if I do a mass import of content for example from the API.

### Motivation

add options for fine tuning

### Detailed Design

add an "EmitEvents" options to collections API.

### Requirements List

I don't know.

### Drawbacks

Option is available in custom operations.

### Alternatives

I don't know.

### Adoption Strategy

No breaking change.

**Comment by br41nslug** [comment]
> with the APIs (rest / graphQL) as well as with the SDK.

The SDK uses the API so both of those would be covered, however i do not agree that the public API should have an option to skip the events considering people may have added extra authentication or data integrity checks which would be a big security/integrity issue if whether to run these was handed over to the end-user. Flows and extensions can skip these events because they are run on the server instead of publicly accessible.

If really needed it is possible to build custom Endpoints/Flows that do not emit events for the specific usecase.

**Reply by rijkvanzanten** [reply]
Fully agree. Being able to skip events on a request-basis feels like a big security vulnerability.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
